lcmgen/emit_cpp.c

```diff
@@ -452,7 +452,7 @@ static void emit_compute_hash(lcmgen_t *lcm, FILE *f, lcm_struct_t *ls)
         emit(0, "{");
         emit(1,     "const __lcm_hash_ptr *fp;");
         emit(1,     "for(fp = p; fp != NULL; fp = fp->parent)");
-        emit(2,         "if(fp->v == %s::getHash)", sn);
+        emit(2,         "if(fp->v == reinterpret_cast<void*>(%s::getHash))", sn);
         emit(3,              "return 0;");
         if(g_ptr_array_size(ls->members)) {
             emit(1, "const __lcm_hash_ptr cp = { p, (void*)%s::getHash };", sn);
```

Comment:
I'm getting a compilation error from line 458 as well.
```ISO C++ forbids casting between pointer-to-function and pointer-to-object```

Using gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4 with `-pedantic -Wall -Wextra -Werror` etc.

I'm wondering why this is a `void *` in the first place. Shouldn't it be possible to have it be an actual function pointer type?